state/stateChanges/writeCollector.go

```diff
@@ -136,6 +103,47 @@ func (scc *stateChangesCollector) getStateChangesForTxs() ([]StateChangesForTx,
 	return stateChangesForTxs, nil
 }
 
+// GetStateChangesForTxs will retrieve the state changes linked with the tx hash.
+func (scc *stateChangesCollector) GetStateChangesForTxs() map[string]*data.StateChanges {
+	scc.stateChangesMut.Lock()
+	defer scc.stateChangesMut.Unlock()
+
+	stateChangesForTxs := make(map[string]*data.StateChanges)
+
+	for _, stateChange := range scc.stateChanges {
+		txHash := string(stateChange.GetTxHash())
+
+		if sc, ok := stateChangesForTxs[txHash]; !ok {
+			stateChangesForTxs[txHash] = &data.StateChanges{StateChanges: []*data.StateChange{
+				{
+					Type:            stateChange.GetType(),
+					Index:           stateChange.GetIndex(),
+					TxHash:          stateChange.GetTxHash(),
+					MainTrieKey:     stateChange.GetMainTrieKey(),
+					MainTrieVal:     stateChange.GetMainTrieVal(),
+					Operation:       stateChange.GetOperation(),
+					DataTrieChanges: stateChange.GetDataTrieChanges(),
+				},
+			},
+			}
+		} else {
+			stateChangesForTxs[txHash].StateChanges = append(sc.StateChanges,
```

Comment:
here we should not append to the sc.StateChanges, but to the stateChangesForTxs[txHash].StateChanges
then we can have the RWMutex as well.

Reply:
can you also add some unit tests for GetStateChangesForTxs ?

Reply:
oh, you are right. Sorry for that, adding unit test as well.

Reply:
but it is very confusing as it is. Will use he same variable name for both.